actionpack/lib/action_controller/metal/http_authentication.rb

```diff
@@ -211,9 +211,9 @@ def request_http_digest_authentication(realm = "Application", message = nil)
         end
       end
 
-      # Returns false on a valid response, true otherwise.
+      # Returns true on a valid response, false otherwise.
       def authenticate(request, realm, &password_procedure)
-        request.authorization && validate_digest_response(request, realm, &password_procedure)
+        request.authorization.present? && validate_digest_response(request, realm, &password_procedure)
```

Comment:
```suggestion
        request.authorization && validate_digest_response(request, realm, &password_procedure)
```
https://guides.rubyonrails.org/api_documentation_guidelines.html#booleans